processor/src/main/java/org/mapstruct/ap/internal/model/source/MappingMethodOptions.java

```diff
@@ -355,6 +355,27 @@ public void applyIgnoreAll(SourceMethod method, TypeFactory typeFactory,
         }
     }
 
+    public void applyIgnoreTargets(SourceMethod method, FormattingMessager messager,
+                                    List<String> ignoreTargets ) {
+        for ( MappingOptions mapping : mappings ) {
+            if ( !mapping.isIgnored() && ignoreTargets.contains( mapping.getTargetName() ) ) {
+                messager.printMessage(
+                        method.getExecutable(),
+                        getBeanMapping().getMirror(),
+                        Message.BEANMAPPING_IGNORE_TARGETS_WITH_MAPPING_TARGET_ERROR,
+                        ignoreTargets,
+                        mapping.getTargetName()
+                );
+            }
+        }
+
+        for (String ignoreTarget : ignoreTargets) {
+            MappingOptions mapping =
```

Comment:
Here we should validate whether `ignoreTarget` is `.` and then provide an error similar to `BEANMAPPING_IGNORE_BY_DEFAULT_WITH_MAPPING_TARGET_THIS`